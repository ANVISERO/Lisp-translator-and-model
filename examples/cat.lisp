(
    ( defvar r )
    ( defvar a 0 )
    ( defvar b 1 )
    (
        loop ( < a b )
        (
            ( setq r ( read ) )
            ( print r )
        )
    )
)