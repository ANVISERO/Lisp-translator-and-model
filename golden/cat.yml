in_source: |-
  ( defvar r )
  ( loop
      ( setq r ( read ) )
      ( format t r )
  )
in_stdin: |-
  foo
out_code: |-
  [{"opcode": "movv", "arg": ["$0", " "]},
   {"opcode": "movv", "arg": ["$0", "$52"]},
   {"opcode": "movv", "arg": ["$69", "$0"]},
   {"opcode": "jp", "arg": [1]},
   {"opcode": "halt"}]
out_stdout: |
  source LoC: 19 code instr: 5
  ============================================================
  foo
  instr_counter: 11  ticks: 57
out_log: |
  DEBUG   machine:simulation    {TICK: 0, PC: 0, ADDR: 0, MEM_OUT: 0, AC: 0, DR: 0} movv ['$0', ' ']
  DEBUG   machine:tick          {TICK: 1, PC: 0, ADDR: 0, MEM_OUT: 0, AC: 0, DR: 0} movv ['$0', ' ']
  DEBUG   machine:tick          {TICK: 2, PC: 0, ADDR: 0, MEM_OUT: 0, AC: 0, DR: 32} movv ['$0', ' ']
  DEBUG   machine:tick          {TICK: 3, PC: 0, ADDR: 0, MEM_OUT: 0, AC: 32, DR: 32} movv ['$0', ' ']
  DEBUG   machine:write         input: ' '
  DEBUG   machine:tick          {TICK: 4, PC: 0, ADDR: 0, MEM_OUT: 32, AC: 32, DR: 32} movv ['$0', ' ']
  DEBUG   machine:tick          {TICK: 5, PC: 1, ADDR: 52, MEM_OUT: 102, AC: 32, DR: 32} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 6, PC: 1, ADDR: 52, MEM_OUT: 102, AC: 32, DR: 102} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 7, PC: 1, ADDR: 0, MEM_OUT: 32, AC: 32, DR: 102} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 8, PC: 1, ADDR: 0, MEM_OUT: 32, AC: 32, DR: 102} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 9, PC: 1, ADDR: 0, MEM_OUT: 32, AC: 102, DR: 102} movv ['$0', '$52']
  DEBUG   machine:write         input: 'f'
  DEBUG   machine:tick          {TICK: 10, PC: 1, ADDR: 0, MEM_OUT: 102, AC: 102, DR: 102} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 11, PC: 2, ADDR: 0, MEM_OUT: 102, AC: 102, DR: 102} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 12, PC: 2, ADDR: 0, MEM_OUT: 102, AC: 102, DR: 102} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 13, PC: 2, ADDR: 69, MEM_OUT: 0, AC: 102, DR: 102} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 14, PC: 2, ADDR: 69, MEM_OUT: 0, AC: 102, DR: 102} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 15, PC: 2, ADDR: 69, MEM_OUT: 0, AC: 102, DR: 102} movv ['$69', '$0']
  DEBUG   machine:write         output: '' << 'f'
  DEBUG   machine:tick          {TICK: 16, PC: 2, ADDR: 69, MEM_OUT: 102, AC: 102, DR: 102} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 17, PC: 3, ADDR: 69, MEM_OUT: 102, AC: 102, DR: 102} jp [1]
  DEBUG   machine:tick          {TICK: 18, PC: 1, ADDR: 69, MEM_OUT: 102, AC: 102, DR: 102} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 19, PC: 1, ADDR: 53, MEM_OUT: 111, AC: 102, DR: 102} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 20, PC: 1, ADDR: 53, MEM_OUT: 111, AC: 102, DR: 111} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 21, PC: 1, ADDR: 0, MEM_OUT: 102, AC: 102, DR: 111} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 22, PC: 1, ADDR: 0, MEM_OUT: 102, AC: 102, DR: 111} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 23, PC: 1, ADDR: 0, MEM_OUT: 102, AC: 111, DR: 111} movv ['$0', '$52']
  DEBUG   machine:write         input: 'o'
  DEBUG   machine:tick          {TICK: 24, PC: 1, ADDR: 0, MEM_OUT: 111, AC: 111, DR: 111} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 25, PC: 2, ADDR: 0, MEM_OUT: 111, AC: 111, DR: 111} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 26, PC: 2, ADDR: 0, MEM_OUT: 111, AC: 111, DR: 111} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 27, PC: 2, ADDR: 70, MEM_OUT: 0, AC: 111, DR: 111} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 28, PC: 2, ADDR: 70, MEM_OUT: 0, AC: 111, DR: 111} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 29, PC: 2, ADDR: 70, MEM_OUT: 0, AC: 111, DR: 111} movv ['$69', '$0']
  DEBUG   machine:write         output: 'f' << 'o'
  DEBUG   machine:tick          {TICK: 30, PC: 2, ADDR: 70, MEM_OUT: 111, AC: 111, DR: 111} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 31, PC: 3, ADDR: 70, MEM_OUT: 111, AC: 111, DR: 111} jp [1]
  DEBUG   machine:tick          {TICK: 32, PC: 1, ADDR: 70, MEM_OUT: 111, AC: 111, DR: 111} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 33, PC: 1, ADDR: 54, MEM_OUT: 111, AC: 111, DR: 111} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 34, PC: 1, ADDR: 54, MEM_OUT: 111, AC: 111, DR: 111} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 35, PC: 1, ADDR: 0, MEM_OUT: 111, AC: 111, DR: 111} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 36, PC: 1, ADDR: 0, MEM_OUT: 111, AC: 111, DR: 111} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 37, PC: 1, ADDR: 0, MEM_OUT: 111, AC: 111, DR: 111} movv ['$0', '$52']
  DEBUG   machine:write         input: 'o'
  DEBUG   machine:tick          {TICK: 38, PC: 1, ADDR: 0, MEM_OUT: 111, AC: 111, DR: 111} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 39, PC: 2, ADDR: 0, MEM_OUT: 111, AC: 111, DR: 111} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 40, PC: 2, ADDR: 0, MEM_OUT: 111, AC: 111, DR: 111} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 41, PC: 2, ADDR: 71, MEM_OUT: 0, AC: 111, DR: 111} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 42, PC: 2, ADDR: 71, MEM_OUT: 0, AC: 111, DR: 111} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 43, PC: 2, ADDR: 71, MEM_OUT: 0, AC: 111, DR: 111} movv ['$69', '$0']
  DEBUG   machine:write         output: 'fo' << 'o'
  DEBUG   machine:tick          {TICK: 44, PC: 2, ADDR: 71, MEM_OUT: 111, AC: 111, DR: 111} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 45, PC: 3, ADDR: 71, MEM_OUT: 111, AC: 111, DR: 111} jp [1]
  DEBUG   machine:tick          {TICK: 46, PC: 1, ADDR: 71, MEM_OUT: 111, AC: 111, DR: 111} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 47, PC: 1, ADDR: 55, MEM_OUT: 0, AC: 111, DR: 111} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 48, PC: 1, ADDR: 55, MEM_OUT: 0, AC: 111, DR: 0} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 49, PC: 1, ADDR: 0, MEM_OUT: 111, AC: 111, DR: 0} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 50, PC: 1, ADDR: 0, MEM_OUT: 111, AC: 111, DR: 0} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 51, PC: 1, ADDR: 0, MEM_OUT: 111, AC: 0, DR: 0} movv ['$0', '$52']
  DEBUG   machine:write         input: '\x00'
  DEBUG   machine:tick          {TICK: 52, PC: 1, ADDR: 0, MEM_OUT: 0, AC: 0, DR: 0} movv ['$0', '$52']
  DEBUG   machine:tick          {TICK: 53, PC: 2, ADDR: 0, MEM_OUT: 0, AC: 0, DR: 0} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 54, PC: 2, ADDR: 0, MEM_OUT: 0, AC: 0, DR: 0} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 55, PC: 2, ADDR: 72, MEM_OUT: 0, AC: 0, DR: 0} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 56, PC: 2, ADDR: 72, MEM_OUT: 0, AC: 0, DR: 0} movv ['$69', '$0']
  DEBUG   machine:tick          {TICK: 57, PC: 2, ADDR: 72, MEM_OUT: 0, AC: 0, DR: 0} movv ['$69', '$0']
  INFO    machine:write         input_buffer: < foo
  WARNING machine:simulation    Input buffer is empty!
  INFO    machine:simulation    output_buffer: > 'foo'
